File: app/src/main/java/com/wkz/viewpager/transformer/FlipHorizontalTransformer.java
package com.wkz.viewpager.transformer;

import android.view.View;

import com.nineoldandroids.view.ViewHelper;

/**
 * @desc 水平快速翻转的转换效果
 * @date 2017-06-30
 */
public class FlipHorizontalTransformer extends AbstractBaseTransformer {

    @Override
    protected void onTransform(View page, float position) {
        float rotation = 180.0F * position;
        ViewHelper.setAlpha(page, rotation > 90.0F || rotation < -90.0F ? 0F : 1F);
        ViewHelper.setPivotX(page, page.getWidth() * 0.5F);
        ViewHelper.setPivotY(page, page.getHeight() * 0.5F);
        ViewHelper.setRotationY(page, rotation);
    }
}

File: view-pager-transformer/src/main/java/com/fphoenixcorneae/transformer/FlipHorizontalTransformer.kt
package com.fphoenixcorneae.transformer

import android.view.View
import com.nineoldandroids.view.ViewHelper

/**
 * @desc 水平快速翻转的转换效果
 * @date 2017-06-30
 */
class FlipHorizontalTransformer : AbstractBaseTransformer() {
    override fun onTransform(page: View, position: Float) {
        val rotation = 180.0f * position
        ViewHelper.setAlpha(page, if (rotation > 90.0f || rotation < -90.0f) 0f else 1f)
        ViewHelper.setPivotX(page, page.width * 0.5f)
        ViewHelper.setPivotY(page, page.height * 0.5f)
        ViewHelper.setRotationY(page, rotation)
    }
}